src/serverlist.cpp

```diff
@@ -93,43 +155,59 @@ CServerListManager::CServerListManager ( const quint16  iNPortNum,
         ThisServerListEntry.strCity = slServInfoSeparateParams[1].left ( MAX_LEN_SERVER_CITY );
 
         // [this server country as QLocale ID]
-        const int iCountry = slServInfoSeparateParams[2].toInt();
-
-        if ( !slServInfoSeparateParams[2].isEmpty() && ( iCountry >= 0 ) && ( iCountry <= QLocale::LastCountry ) )
+        bool      ok;
+        const int iCountry = slServInfoSeparateParams[2].toInt ( &ok );
+        if ( ok && iCountry >= 0 && iCountry <= QLocale::LastCountry )
         {
             ThisServerListEntry.eCountry = static_cast<QLocale::Country> ( iCountry );
         }
     }
 
+    // per definition, the very first entry is this server and this entry will
```

Comment:
... is this server (maybe better: this directory server?) Why is this comment here, not below ServerList.clear()? Without having looked at what ServerList.clear() does, I assume it clears the server list and ServerList.append ( ThisServerListEntry ) adds the directory server entry?

Reply:
Yes.